src/test/java/com/es/phoneshop/model/product/ProductSearchPredicateTest.java

```diff
@@ -0,0 +1,66 @@
+package com.es.phoneshop.model.product;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ProductSearchPredicateTest {
+    private ProductSearchPredicate predicate;
+    private Product defaultProduct;
+
+    @Before
+    public void setup() {
+        defaultProduct = new Product();
+    }
+
+    @Test
+    public void testProductThatContainSearchMustBeFound() {
+        String query = "Samsung";
+        predicate = new ProductSearchPredicate(query);
+        defaultProduct.setDescription("Samsung Galaxy S");
+
+        boolean result = predicate.test(defaultProduct);
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void testProductWithEmptyDescriptionMustNotBeFound() {
+        String query = "Samsung";
+        predicate = new ProductSearchPredicate(query);
+        defaultProduct.setDescription("");
+
+        boolean result = predicate.test(defaultProduct);
+
+        assertFalse(result);
+    }
+
+    @Test
+    public void testEmptyQueryFoundAllProducts() {
+        String query = "";
+        predicate = new ProductSearchPredicate(query);
+
+        defaultProduct.setDescription("Hello");
```

Comment:
Test context preparation should be placed into "when" part of the test or into setUp() method